integration-cli/docker_cli_build_test.go

```diff
@@ -4694,3 +4694,92 @@ func TestBuildDockerfileOutsideContext(t *testing.T) {
 
 	logDone("build - Dockerfile outside context")
 }
+
+func TestBuildSpaces(t *testing.T) {
+	// Test to make sure that leading/trailing spaces on a command
+	// doesn't change the error msg we get
+	var (
+		err1 error
+		err2 error
+	)
+
+	name := "testspaces"
+	defer deleteImages(name)
+	ctx, err := fakeContext("FROM busybox\nRUN\n",
+		map[string]string{
+			"Dockerfile": "FROM busybox\nRUN\n",
+		})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Close()
+
+	timeRE := regexp.MustCompile(`time="[^Z]*Z"`)
+
+	if _, err1 = buildImageFromContext(name, ctx, false); err1 == nil {
+		t.Fatal("Build 1 was supposed to fail, but didn't")
+	}
+
+	ctx.Add("Dockerfile", "FROM busybox\nRUN    ")
+	if _, err2 = buildImageFromContext(name, ctx, false); err2 == nil {
+		t.Fatal("Build 2 was supposed to fail, but didn't")
+	}
+
+	// Mask the times
+	e1 := timeRE.ReplaceAllString(err1.Error(), "time")
+	e2 := timeRE.ReplaceAllString(err2.Error(), "time")
+
+	if strings.Replace(e1, " ", "", -1) != strings.Replace(e2, " ", "", -1) {
+		t.Fatalf("Build 2's error wasn't the same as build 1's\n1:%s\n2:%s", err1, err2)
+	}
+
+	ctx.Add("Dockerfile", "FROM busybox\n   RUN")
+	if _, err2 = buildImageFromContext(name, ctx, false); err2 == nil {
+		t.Fatal("Build 3 was supposed to fail, but didn't")
+	}
+
+	// Mask the times
+	e1 = timeRE.ReplaceAllString(err1.Error(), "time")
+	e2 = timeRE.ReplaceAllString(err2.Error(), "time")
+
+	if strings.Replace(e1, " ", "", -1) != strings.Replace(e2, " ", "", -1) {
+		t.Fatalf("Build 3's error wasn't the same as build 1's\n1:%s\n3:%s", err1, err2)
+	}
+
+	ctx.Add("Dockerfile", "FROM busybox\n   RUN    ")
+	if _, err2 = buildImageFromContext(name, ctx, false); err2 == nil {
+		t.Fatal("Build 4 was supposed to fail, but didn't")
+	}
+
+	// Mask the times
+	e1 = timeRE.ReplaceAllString(err1.Error(), "time")
+	e2 = timeRE.ReplaceAllString(err2.Error(), "time")
+
+	if strings.Replace(e1, " ", "", -1) != strings.Replace(e2, " ", "", -1) {
+		t.Fatalf("Build 4's error wasn't the same as build 1's\n1:%s\n4:%s", err1, err2)
+	}
+
+	logDone("build - test spaces")
+}
+
+func TestBuildSpacesWithQuotes(t *testing.T) {
```

Comment:
Why should it fail on master?  I didn't actually introduce new behavior in this PR, I'm just adding tests to make sure no one breaks/changes the behavior in the future.  At least that's what I think I was doing  ;-)  22 days ago is a lifetime.